step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/resolvedplan/ResolvedPlanBuilder.java

```diff
@@ -78,7 +81,7 @@ public <ARTEFACT extends AbstractArtefact> void checkAndAddMissingResolvedPlanNo
             resolvedPlanCache.computeIfAbsent(artefactHash, n -> {
                 AbstractArtefact artefactClone = dynamicBeanResolver.cloneDynamicValues(artefact);
                 String parentPlanNodeId = findClosestParentNodeId(parentReportNode, reportNodeCache);
-                ResolvedPlanNode resolvedPlanNode = new ResolvedPlanNode(artefactClone, artefactHash, parentPlanNodeId, parentSource);
```

Comment:
What happens if
1) we already have a child at position 0
2) we have more than one dynamic calls i.e. more than one element to add at position 0?